// @oracle/oraclejet-preact/cjs/resources/nls/uk/bundle.js
/* @oracle/oraclejet-preact: 14.1.0 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const bundle = {
    close: () => 'Закрити',
    error: () => 'Помилка',
    warn: () => 'Попередження',
    info: () => 'Відомості',
    confirmation: () => 'Підтвердження',
    progressIndeterminate: () => 'Триває',
    indexer_separatorLabel: (p) => 'Від ' + p['FROM_SECTION'] + ' до ' + p['TO_SECTION'] + '.',
    indexer_disabledLabel: (p) => p['SECTION'] + '. Не знайдено відповідних розділів.',
    indexer_othersLabel: () => '#',
    indexer_keyboardInstructionText: () => 'Натисніть клавішу Enter, щоб вибрати значення.',
    indexer_touchInstructionText: () => 'Двічі торкніться й утримайте, щоб увійти в режим жестів, а потім перетягніть угору або вниз, щоб змінити значення.',
    inputPassword_show: () => 'Показати пароль',
    inputPassword_hide: () => 'Приховати пароль',
    inputPassword_hidden: () => 'Пароль приховано',
    formControl_clear: () => 'Очистити',
    formControl_loading: () => 'Завантаження',
    formControl_maxLengthExceeded: (p) => 'Максимальну довжину (' + p['MAX_LENGTH'] + ') перевищено.',
    formControl_maxLengthRemaining: (p) => 'Залишилася така кількість символів: ' + p['CHARACTER_COUNT'] + '.',
    formControl_requiredMessageDetail: () => 'Введіть значення.',
    select_requiredMessageDetail: () => 'Виберіть значення.',
    select_noMatchesFound: () => 'Збігів не знайдено.',
    select_oneMatchFound: () => 'Знайдено один збіг',
    select_sizeMatchesFound: (p) => 'Знайдено таку кількість збігів: ' + p['TOTAL_SIZE'],
    select_sizeOrMoreMatchesFound: (p) => 'Знайдено ' + p['TOTAL_SIZE'] + ' або більше збігів',
    selectMultiple_showSelectedValues: () => 'Показувати лише вибрані значення.',
    selectMultiple_valuesSelected: (p) => 'Вибрано таку кількість значень: ' + p['VALUE_COUNT'] + '.',
    selectMultiple_countPlus: (p) => p['COUNT'] + '+',
    selector_selected: () => 'Прапорець установлено',
    selector_unselected: () => 'Прапорець не встановлено',
    userAssistance_learnMore: () => 'Додаткові відомості',
    userAssistance_required: () => "Обов'язково",
    filePicker_singleFileUploadError: () => 'Передайте один файл за раз',
    filePicker_singleTypeUploadError: (p) => 'Неможливо передати файл типу ' + p['fileType'],
    filePicker_multipleFileTypeUploadError: (p) => 'Неможливо передати файли типів ' + p['fileTypes'],
    filePicker_unknownFileTypeUploadError: () => 'невідомо',
    filePicker_dropzonePrimaryText: () => 'Перетягнути',
    filePicker_dropzoneSecondaryText: () => 'Виберіть файл або перетягніть його сюди',
    filePicker_dropzoneSecondaryTextMultiple: () => 'Виберіть файли або перетягніть їх сюди',
    filePicker_addFiles: () => 'Додати файли',
    plural_separator: () => ', ',
    inputNumber_increase: () => 'Збільшення',
    inputNumber_decrease: () => 'Зменшення',
    inputNumber_converterParseError: () => 'Введіть число.',
    noData_message: () => 'Немає позицій для відображення',
    message_navigationFromMessagesRegion: () => 'Вхід в область повідомлень. Натисніть клавішу F6, щоб повернутися до елемента, на якому раніше було встановлено фокус.',
    message_navigationToMessagesRegion: () => 'В області повідомлень наявні нові повідомлення. Натисніть клавішу F6, щоб перейти до останньої області повідомлень.',
    messageToast_allMessagesClosed: () => 'Усі спливні повідомлення закрито.'
};

exports["default"] = bundle;
/*  */
