// @oracle/oraclejet-preact/es/UNSAFE_ComponentMessage.js
/* @oracle/oraclejet-preact: 14.1.0 */
import { jsxs, jsx } from 'preact/jsx-runtime';
import "./UNSAFE_ComponentMessage.css";
import { useTranslationBundle } from './hooks/UNSAFE_useTranslationBundle.js';
import { severityBasedStyleClass, isSeverityIconNeeded, MessageStartIcon, MessageSummary, MessagesManager } from './PRIVATE_Message.js';
import { HiddenAccessible } from './UNSAFE_HiddenAccessible.js';
import { classNames } from './utils/UNSAFE_classNames.js';
import 'preact/hooks';
import './UNSAFE_Environment.js';
import 'preact';
import './UNSAFE_Layer.js';
import 'preact/compat';
import './UNSAFE_Flex.js';
import './tslib.es6-821b2d99.js';
import './utils/UNSAFE_interpolations/dimensions.js';
import './utils/UNSAFE_arrayUtils.js';
import './utils/UNSAFE_size.js';
import './_curry1-ab762f3c.js';
import './utils/UNSAFE_mergeInterpolations.js';
import './_curry3-77af75f1.js';
import './_curry2-9cf34233.js';
import './_has-e54ed87c.js';
import './utils/UNSAFE_interpolations/boxalignment.js';
import './keys-6f2f2841.js';
import './utils/UNSAFE_interpolations/flexbox.js';
import './utils/UNSAFE_interpolations/flexitem.js';
import './utils/PRIVATE_timer.js';
import './UNSAFE_Button.js';
import './UNSAFE_BaseButton.js';
import './hooks/UNSAFE_usePress.js';
import './hooks/UNSAFE_useHover.js';
import './hooks/UNSAFE_useToggle.js';
import './hooks/UNSAFE_useActive.js';
import './utils/PRIVATE_clientHints.js';
import './hooks/UNSAFE_useTabbableMode.js';
import './utils/UNSAFE_mergeProps.js';
import './UNSAFE_ThemedIcons.js';
import './UNSAFE_Icons.js';
import './UNSAFE_Icon.js';
import './hooks/UNSAFE_useTooltip.js';
import './hooks/UNSAFE_useId.js';
import './UNSAFE_Floating.js';
import './index-8cf878d9.js';
import './hooks/UNSAFE_useUser.js';
import './utils/PRIVATE_floatingUtils.js';
import './utils/PRIVATE_refUtils.js';
import './hooks/UNSAFE_useOutsideClick.js';
import './hooks/UNSAFE_useFocus.js';
import './hooks/UNSAFE_useTouch.js';
import './hooks/UNSAFE_useAnimation.js';
import './hooks/UNSAFE_useTheme.js';
import './utils/UNSAFE_getLocale.js';
import './utils/UNSAFE_stringUtils.js';
import './utils/UNSAFE_logger.js';
import './utils/UNSAFE_soundUtils.js';
import './PRIVATE_TransitionGroup.js';
import './hooks/UNSAFE_useMessagesContext.js';

const baseStyles = "_11n5zgz";
/**
 * The component that renders an individual message for inline component messaging.
 */

function ComponentMessage({
  detail,
  fieldLabel,
  severity = 'error'
}) {
  const classes = classNames([baseStyles, severityBasedStyleClass(severity, 'inline')]);
  const translations = useTranslationBundle('@oracle/oraclejet-preact');
  const severityTranslations = {
    confirmation: translations.confirmation(),
    error: translations.error(),
    info: translations.info(),
    warning: translations.warn()
  };
  return jsxs("div", Object.assign({
    class: classes,
    role: "alert",
    "aria-atomic": "true"
  }, {
    children: [fieldLabel && jsx(HiddenAccessible, {
      children: fieldLabel
    }), isSeverityIconNeeded(severity) && jsx(MessageStartIcon, {
      variant: "inline",
      severity: severity,
      translations: severityTranslations
    }), jsx(MessageSummary, {
      variant: "inline",
      text: detail
    })]
  }));
}

const rootStyles = "_4wirz6";
/**
 * Converts the messages data into Item
 *
 * @param messages The messages data passed down as prop
 * @returns The messages data as Item
 */

function generateMessagesData(messages) {
  return messages.map((message, index) => {
    return {
      key: index,
      data: {
        closeAffordance: 'off',
        severity: message.severity || 'error',
        detail: message.detail
      }
    };
  });
}

function ComponentMessageContainer({
  fieldLabel,
  messages = []
}) {
  return jsx("div", Object.assign({
    class: rootStyles
  }, {
    children: jsx(MessagesManager, Object.assign({
      data: generateMessagesData(messages)
    }, {
      children: ({
        item
      }) => jsx(ComponentMessage, {
        detail: item.data.detail,
        fieldLabel: fieldLabel,
        severity: item.data.severity
      }, item.key)
    }))
  }));
}

export { ComponentMessage, ComponentMessageContainer };
/*  */
