// @oracle/oraclejet-preact/es/UNSAFE_Heading.js
/* @oracle/oraclejet-preact: 14.1.0 */
import { _ as __rest } from './tslib.es6-821b2d99.js';
import { jsx } from 'preact/jsx-runtime';
import { k as keys_1 } from './keys-6f2f2841.js';
import { stringLiteralArray } from './utils/UNSAFE_arrayUtils.js';
import "./UNSAFE_Heading.css";
import { mergeInterpolations } from './utils/UNSAFE_mergeInterpolations.js';
import './_curry1-ab762f3c.js';
import './_has-e54ed87c.js';
import './utils/UNSAFE_classNames.js';
import './_curry3-77af75f1.js';
import './_curry2-9cf34233.js';

const headingElementTypes = stringLiteralArray(['h1', 'h2', 'h3', 'h4', 'h5', 'h6']);
const headingsizeMap = {
  h1: '2xl',
  h2: 'xl',
  h3: 'lg',
  h4: 'md',
  h5: 'sm',
  h6: 'xs'
};
const styles = {
  base: "ubmk3n",
  size: {
    xs: "augrqi",
    sm: "_24o9pl",
    md: "_1k987i8",
    lg: "_1c133",
    xl: "_1euwonb",
    '2xl': "_46c4ix",
    inherit: "_1vgm62q"
  },
  childrenMargins: "_1pdnu5a"
};
const sizeOptions = keys_1(styles.size);
const interpolations = [() => ({
  class: styles.base
}), ({
  size = '2xl'
}) => ({
  class: styles.size[size]
})];
const styleInterpolations = mergeInterpolations(interpolations);
function Heading(_a) {
  var _b;

  var {
    children
  } = _a,
      props = __rest(_a, ["children"]);

  const Component = props.as; //@ts-ignore

  props.size = (_b = props.size) !== null && _b !== void 0 ? _b : headingsizeMap[Component];

  const _c = styleInterpolations(props),
        {
    class: cls
  } = _c,
        style = __rest(_c, ["class"]);

  return jsx(Component, Object.assign({
    className: cls,
    style: style,
    id: props.id
  }, {
    children: jsx("span", Object.assign({
      className: styles.childrenMargins
    }, {
      children: children
    }))
  }));
}

export { Heading, headingElementTypes, sizeOptions };
/*  */
