// @oracle/oraclejet-preact/es/UNSAFE_TextArea.js
/* @oracle/oraclejet-preact: 14.1.0 */
import { jsx, jsxs } from 'preact/jsx-runtime';
import { Fragment } from 'preact';
import { forwardRef, useCallback as useCallback$1 } from 'preact/compat';
import { useFormContext } from './hooks/UNSAFE_useFormContext.js';
import { useFormFieldContext, FormFieldContext } from './hooks/UNSAFE_useFormFieldContext.js';
import { useFocusableTextField } from './hooks/UNSAFE_useFocusableTextField.js';
import { Label } from './UNSAFE_Label.js';
import { ReadonlyTextField, ReadonlyTextFieldInput, TextFieldInput, MaxLengthLiveRegion, TextField } from './UNSAFE_TextField.js';
import { useTextField } from './hooks/UNSAFE_useTextField.js';
import { InlineUserAssistance } from './UNSAFE_UserAssistance.js';
import { useLengthFilter } from './hooks/UNSAFE_useLengthFilter.js';
import "./UNSAFE_TextArea.css";
import { classNames } from './utils/UNSAFE_classNames.js';
import { useCallback } from 'preact/hooks';
import { usePress } from './hooks/UNSAFE_usePress.js';
import { useCurrentValueReducer } from './hooks/UNSAFE_useCurrentValueReducer.js';
import './tslib.es6-821b2d99.js';
import './hooks/UNSAFE_useFocusWithin.js';
import './hooks/UNSAFE_useTabbableMode.js';
import './utils/UNSAFE_interpolations/text.js';
import './keys-6f2f2841.js';
import './_curry1-ab762f3c.js';
import './_has-e54ed87c.js';
import './utils/UNSAFE_mergeInterpolations.js';
import './_curry3-77af75f1.js';
import './_curry2-9cf34233.js';
import './hooks/UNSAFE_useTranslationBundle.js';
import './UNSAFE_Environment.js';
import './UNSAFE_Layer.js';
import './UNSAFE_LabelValueLayout.js';
import './UNSAFE_Flex.js';
import './utils/UNSAFE_interpolations/dimensions.js';
import './utils/UNSAFE_arrayUtils.js';
import './utils/UNSAFE_size.js';
import './utils/UNSAFE_interpolations/boxalignment.js';
import './utils/UNSAFE_interpolations/flexbox.js';
import './utils/UNSAFE_interpolations/flexitem.js';
import './hooks/UNSAFE_useTextFieldInputHandlers.js';
import './utils/PRIVATE_clientHints.js';
import './hooks/UNSAFE_useDebounce.js';
import './UNSAFE_LiveRegion.js';
import './hooks/UNSAFE_useId.js';
import './UNSAFE_ComponentMessage.js';
import './PRIVATE_Message.js';
import './utils/PRIVATE_timer.js';
import './UNSAFE_Button.js';
import './UNSAFE_BaseButton.js';
import './hooks/UNSAFE_useHover.js';
import './hooks/UNSAFE_useToggle.js';
import './hooks/UNSAFE_useActive.js';
import './utils/UNSAFE_mergeProps.js';
import './UNSAFE_ThemedIcons.js';
import './UNSAFE_Icons.js';
import './UNSAFE_Icon.js';
import './hooks/UNSAFE_useTooltip.js';
import './UNSAFE_Floating.js';
import './index-8cf878d9.js';
import './hooks/UNSAFE_useUser.js';
import './utils/PRIVATE_floatingUtils.js';
import './utils/PRIVATE_refUtils.js';
import './hooks/UNSAFE_useOutsideClick.js';
import './hooks/UNSAFE_useFocus.js';
import './hooks/UNSAFE_useTouch.js';
import './hooks/UNSAFE_useAnimation.js';
import './hooks/UNSAFE_useTheme.js';
import './utils/UNSAFE_getLocale.js';
import './utils/UNSAFE_stringUtils.js';
import './utils/UNSAFE_logger.js';
import './utils/UNSAFE_soundUtils.js';
import './PRIVATE_TransitionGroup.js';
import './hooks/UNSAFE_useMessagesContext.js';
import './UNSAFE_HiddenAccessible.js';
import './utils/UNSAFE_lengthFilter.js';

const rootStyles = {
  base: "_6ov0uh"
};
const remainingTextStyles = {
  hide: "bt4itw"
}; // the spans surrounding the {valueLength}, / and {maxlength}  need to be display: inline-block
// so that the order is flipped horizontally in RTL.

const spanStyles = "_1n8efrh";

const noop = () => {};

function MaxLengthCounter({
  maxLength,
  valueLength = 0,
  onClick
}) {
  const {
    isFocused
  } = useFormFieldContext(); // We don't want mousedown on this component to do anything, so prevent the default behavior.

  const killEvent = useCallback(event => {
    event.preventDefault();
  }, []);
  const textClasses = classNames([!isFocused && remainingTextStyles.hide]);
  const {
    pressProps
  } = usePress(onClick !== null && onClick !== void 0 ? onClick : noop);
  return jsx("div", Object.assign({
    class: rootStyles.base,
    onMouseDown: killEvent
  }, pressProps, {
    children: jsxs("span", Object.assign({
      class: textClasses
    }, {
      children: [jsx("span", Object.assign({
        class: spanStyles
      }, {
        children: valueLength
      })), jsx("span", Object.assign({
        class: spanStyles
      }, {
        children: "/"
      })), jsx("span", Object.assign({
        class: spanStyles
      }, {
        children: maxLength
      }))]
    }))
  }));
}

const TextArea = forwardRef(({ ariaDescribedBy, assistiveText, autoComplete = 'off', autoFocus = false, helpSourceLink, helpSourceText, isDisabled: propIsDisabled, isReadonly: propIsReadonly, isRequired = false, isRequiredShown, label, labelEdge: propLabelEdge, labelStartWidth: propLabelStartWidth, maxLength, maxLengthCounter = 'remaining', maxLengthUnit, messages, placeholder, resize, role, rows, textAlign: propTextAlign, userAssistanceDensity: propUserAssistanceDensity, value, variant = 'default', onInput, onCommit }, ref) => {
    const { currentCommitValue, dispatch } = useCurrentValueReducer({ value });
    const onInputAndDispatch = useCallback$1((detail) => {
        // Should dispatch happen first? This will queue up a re-render, ordering should not cause issues (this is async)
        dispatch({ type: 'input', payload: detail.value });
        onInput === null || onInput === void 0 ? void 0 : onInput(detail);
    }, [onInput]);
    const onCommitAndDispatch = useCallback$1((detail) => {
        // Should dispatch happen first? This will queue up a re-render, ordering should not cause issues (this is async)
        dispatch({ type: 'commit', payload: detail.value });
        onCommit === null || onCommit === void 0 ? void 0 : onCommit(detail);
    }, [onCommit]);
    const { isDisabled: isFormDisabled, isReadonly: isFormReadonly, labelEdge: formLabelEdge, labelStartWidth: formLabelStartWidth, textAlign: formTextAlign, userAssistanceDensity: formUserAssistanceDensity } = useFormContext();
    // default to FormContext values if component properties are not specified
    const isDisabled = propIsDisabled !== null && propIsDisabled !== void 0 ? propIsDisabled : isFormDisabled;
    const isReadonly = propIsReadonly !== null && propIsReadonly !== void 0 ? propIsReadonly : isFormReadonly;
    const labelEdge = propLabelEdge !== null && propLabelEdge !== void 0 ? propLabelEdge : formLabelEdge;
    const labelStartWidth = propLabelStartWidth !== null && propLabelStartWidth !== void 0 ? propLabelStartWidth : formLabelStartWidth;
    const textAlign = propTextAlign !== null && propTextAlign !== void 0 ? propTextAlign : formTextAlign;
    const userAssistanceDensity = propUserAssistanceDensity !== null && propUserAssistanceDensity !== void 0 ? propUserAssistanceDensity : formUserAssistanceDensity;
    const { enabledElementRef, readonlyElementRef, focusProps, isFocused } = useFocusableTextField({ isDisabled, isReadonly, ref });
    const { formFieldContext, inputProps, labelProps, textFieldProps, userAssistanceProps } = useTextField({
        ariaDescribedBy,
        contentVariant: 'textarea',
        isDisabled,
        isFocused,
        isReadonly,
        labelEdge,
        messages,
        styleVariant: variant,
        value
    });
    // Handle length filter for user-typed inputs and controlled inputs
    const { isMaxLengthExceeded, valueLength, onFilteredInput } = useLengthFilter({
        maxLength,
        maxLengthUnit,
        value,
        onInput: onInputAndDispatch,
        onCommit: onCommitAndDispatch
    });
    const labelComp = labelEdge !== 'none' ? jsx(Label, Object.assign({}, labelProps, { children: label })) : undefined;
    const fieldLabelProps = {
        label: labelEdge !== 'none' ? labelComp : undefined,
        labelEdge: labelEdge !== 'none' ? labelEdge : undefined,
        labelStartWidth: labelEdge !== 'none' ? labelStartWidth : undefined
    };
    const ariaLabel = labelEdge === 'none' ? label : undefined;
    const inlineUserAssistance = isDisabled || isReadonly ? (
    // save space for user assistance if density is 'efficient', even though we don't
    // render user assistance for disabled or readonly fields
    userAssistanceDensity !== 'efficient' ? undefined : (jsx(InlineUserAssistance, Object.assign({ userAssistanceDensity: userAssistanceDensity }, userAssistanceProps)))) : (jsx(InlineUserAssistance, Object.assign({ assistiveText: assistiveText, fieldLabel: label, helpSourceLink: helpSourceLink, helpSourceText: helpSourceText, messages: messages, isRequiredShown: isRequiredShown, userAssistanceDensity: userAssistanceDensity }, userAssistanceProps)));
    if (isReadonly) {
        // TODO: We need to conditionally render the readonly innerReadonlyField as a textarea or div
        //  also, the div will need to use white-space: pre-wrap so that the text will wrap like textarea
        //  see JET-50636
        return (jsx(FormFieldContext.Provider, Object.assign({ value: formFieldContext }, { children: jsx(ReadonlyTextField, Object.assign({ role: "presentation", inlineUserAssistance: inlineUserAssistance, variant: "textarea" }, fieldLabelProps, { children: jsx(ReadonlyTextFieldInput, { ariaDescribedBy: ariaDescribedBy, ariaLabel: ariaLabel, ariaLabelledBy: labelProps.id, as: "textarea", elementRef: readonlyElementRef, rows: rows, autoFocus: autoFocus, textAlign: textAlign, value: value, variant: "textarea", hasInsideLabel: label !== undefined && labelEdge === 'inside' }) })) })));
    }
    const onClickCounter = useCallback$1(() => {
        var _a, _b;
        const elm = enabledElementRef.current;
        // for setSelectionRange(), len needs to be a number or null.
        const len = elm ? elm.value.length : null;
        // Since the counter element is after the text, clicking in this
        // should put the selection at the end of the text
        (_a = enabledElementRef.current) === null || _a === void 0 ? void 0 : _a.setSelectionRange(len, len);
        // Clicking the counter should put the focus on the input field
        (_b = enabledElementRef.current) === null || _b === void 0 ? void 0 : _b.focus();
    }, []);
    const mainContent = (jsxs(Fragment, { children: [jsx(TextFieldInput, Object.assign({ as: "textarea", ariaLabel: ariaLabel, autoComplete: autoComplete, autoFocus: autoFocus, currentCommitValue: currentCommitValue, hasInsideLabel: labelComp !== undefined && labelEdge === 'inside', isRequired: isRequired, inputRef: enabledElementRef, onCommit: onCommitAndDispatch, onInput: onFilteredInput, placeholder: placeholder, role: role, rows: rows, textAlign: textAlign, value: value }, inputProps)), maxLengthCounter === 'remaining' && maxLength !== undefined && (jsx(MaxLengthCounter, Object.assign({}, { maxLength, valueLength, parentHasFocus: isFocused, onClick: onClickCounter }))), maxLength !== undefined && (jsx(MaxLengthLiveRegion, Object.assign({}, { isMaxLengthExceeded, maxLength, valueLength })))] }));
    return (jsx(FormFieldContext.Provider, Object.assign({ value: formFieldContext }, { children: jsx(TextField, Object.assign({ mainContent: mainContent, inlineUserAssistance: inlineUserAssistance, onBlur: focusProps === null || focusProps === void 0 ? void 0 : focusProps.onfocusout, onFocus: focusProps === null || focusProps === void 0 ? void 0 : focusProps.onfocusin, resize: resize }, textFieldProps, fieldLabelProps)) })));
});

export { MaxLengthCounter, TextArea };
/*  */
