// @oracle/oraclejet-preact/es/UNSAFE_Skeleton.js
/* @oracle/oraclejet-preact: 14.1.0 */
import { _ as __rest } from './tslib.es6-821b2d99.js';
import { jsx } from 'preact/jsx-runtime';
import "./UNSAFE_Skeleton.css";
import { classNames } from './utils/UNSAFE_classNames.js';
import { dimensionInterpolations } from './utils/UNSAFE_interpolations/dimensions.js';
import { mergeInterpolations } from './utils/UNSAFE_mergeInterpolations.js';
import { borderInterpolations } from './utils/UNSAFE_interpolations/borders.js';
import { stringLiteralArray } from './utils/UNSAFE_arrayUtils.js';
import './utils/UNSAFE_size.js';
import './_curry1-ab762f3c.js';
import './_curry3-77af75f1.js';
import './_curry2-9cf34233.js';
import './_has-e54ed87c.js';

const dimensions = stringLiteralArray(['height', 'width']);
const border = stringLiteralArray(['borderRadius']); // Create an array [dimensionInterpolations['height'], dimensionInterpolations['width']]

const skeletonDimensionInterpolation = Array.from(dimensions, x => dimensionInterpolations[x]);
const borderRadiusInterpolation = Array.from(border, x => borderInterpolations[x]);
const interpolations = [...Object.values(borderRadiusInterpolation), ...Object.values(skeletonDimensionInterpolation)];
const SkeletonInterpolations = mergeInterpolations(interpolations);
/* Styling for skeleton main (loop) animation */

const animationSkeletonStyle = {
  base: "lsecum"
};
/**
 * Skeleton component allows the appropriate skeleton to be rendered based on the
 * property values
 **/

function Skeleton(_a) {
  var props = __rest(_a, []);

  const skeletonDimensions = SkeletonInterpolations(Object.assign({
    width: '100%',
    borderRadius: 0
  }, props));
  const itemClasses = classNames([animationSkeletonStyle.base]);
  return jsx("div", {
    style: skeletonDimensions,
    class: itemClasses
  });
}

/**
 * @license
 * Copyright (c) %FIRST_YEAR% %CURRENT_YEAR%, Oracle and/or its affiliates.
 * The Universal Permissive License (UPL), Version 1.0
 * as shown at https://oss.oracle.com/licenses/upl/
 * @ignore
 */

export { Skeleton };
/*  */
