// @oracle/oraclejet-preact/es/UNSAFE_ListItemLayout.js
/* @oracle/oraclejet-preact: 14.1.0 */
import { jsx, jsxs } from 'preact/jsx-runtime';
import "./UNSAFE_ListItemLayout.css";
import { Grid } from './UNSAFE_Grid.js';
import { classNames } from './utils/UNSAFE_classNames.js';
import { Text } from './UNSAFE_Text.js';
import { useTabbableMode } from './hooks/UNSAFE_useTabbableMode.js';
import './tslib.es6-821b2d99.js';
import './utils/UNSAFE_interpolations/grid.js';
import './utils/UNSAFE_interpolations/boxalignment.js';
import './keys-6f2f2841.js';
import './_curry1-ab762f3c.js';
import './_has-e54ed87c.js';
import './utils/UNSAFE_size.js';
import './utils/UNSAFE_arrayUtils.js';
import './utils/UNSAFE_mergeInterpolations.js';
import './_curry3-77af75f1.js';
import './_curry2-9cf34233.js';
import 'preact';
import 'preact/hooks';

const childrenStyles = "_1i0kogh"; // TODO: revisit idea of negative margin when selector has a proper API for handling spacing
// Jira: JET-54543
// negative margin added to selector

const selectorStyles = "_14sujpc"; // most content will be vertically centered

const centerVerticalStyles = "_3lfxww"; // extra spacing is needed between meta, trailing, and action components

const leftPaddingStyles = "do20s5"; // the central text area displays overline, primary, secondary, and tertiary in a column

const flexColumnStyles$1 = "_198d627"; // "extra" components in row 1 and the link in row 2 are justified on the righthand side of the list item layout

const flexRowStyles = "_6482su";
const rightJustifyStyles = "_1tszpd1";
const rowAlignStyles = "_1lxwy0w";
function ListItemLayout({
  children,
  selector,
  leading,
  primary,
  secondary,
  tertiary,
  overline,
  metadata,
  trailing,
  action,
  quaternary,
  navigation
}) {
  // generate grid skeleton, adding cells based on which props are defined
  const {
    columnsTemplate,
    firstCell,
    secondCell,
    lastCell,
    hasSecondRow
  } = getColumnsTemplate(selector, leading, metadata, trailing, action, quaternary, navigation); // if ListItemLayout component has a child (ListItemTextLayout) then render that instead of individual props

  const textArea = children ? jsx("div", Object.assign({
    className: childrenStyles
  }, {
    children: children
  })) : jsxs("div", Object.assign({
    className: flexColumnStyles$1
  }, {
    children: [jsx("div", {
      children: overline
    }), jsx("div", {
      children: primary
    }), jsx("div", {
      children: secondary
    }), jsx("div", {
      children: tertiary
    })]
  })); // a second row is needed when quaternary or navigation are defined

  const secondRow = hasSecondRow ? [firstCell && jsx("span", {
    children: " "
  }), secondCell && jsx("span", {
    children: " "
  }), jsx("div", Object.assign({
    className: rowAlignStyles
  }, {
    children: quaternary
  })), lastCell && jsx("div", Object.assign({
    className: rightJustifyStyles
  }, {
    children: navigation
  }))] : null;
  return jsxs(Grid, Object.assign({
    gap: "4x",
    gridTemplateColumns: columnsTemplate
  }, {
    children: [firstCell, secondCell, jsx("div", Object.assign({
      className: centerVerticalStyles
    }, {
      children: textArea
    })), lastCell, secondRow]
  }));
}

function getColumnsTemplate(selector, leading, metadata, trailing, action, quaternary, navigation) {
  const columnsTemplateBuilder = [];
  let firstCell = null;

  if (selector) {
    columnsTemplateBuilder.push('auto');
    firstCell = jsx("div", Object.assign({
      className: selectorStyles
    }, {
      children: selector
    }));
  }

  let secondCell = null;

  if (leading) {
    columnsTemplateBuilder.push('auto');
    secondCell = jsx("div", Object.assign({
      className: centerVerticalStyles
    }, {
      children: leading
    }));
  }

  columnsTemplateBuilder.push('1fr'); // Text area always gets 1 full fraction of allotted space

  let lastCell = null;

  if (metadata || trailing || action || navigation) {
    columnsTemplateBuilder.push('auto');
    lastCell = jsxs("div", Object.assign({
      className: flexRowStyles
    }, {
      children: [jsx("div", Object.assign({
        className: classNames([centerVerticalStyles, metadata ? leftPaddingStyles : ''])
      }, {
        children: metadata
      })), jsx("div", Object.assign({
        className: classNames([centerVerticalStyles, trailing ? leftPaddingStyles : ''])
      }, {
        children: trailing
      })), jsx("div", Object.assign({
        className: classNames([centerVerticalStyles, action ? leftPaddingStyles : ''])
      }, {
        children: action
      }))]
    }));
  }

  const columnsTemplate = columnsTemplateBuilder.join(' ');
  const hasSecondRow = quaternary || navigation ? true : false;
  return {
    columnsTemplate,
    firstCell,
    secondCell,
    lastCell,
    hasSecondRow
  };
}

const textVariants = {
    primary: 'primary',
    secondary: 'primary',
    tertiary: 'secondary',
    quaternary: 'secondary',
    overline: 'secondary',
    metadata: 'secondary'
};
const textSizes = {
    primary: 'md',
    secondary: 'sm',
    tertiary: 'xs',
    quaternary: 'sm',
    overline: 'xs',
    metadata: 'sm'
};
const textLineClamps = {
    primary: undefined,
    secondary: 2,
    tertiary: 2,
    quaternary: 2,
    overline: 1,
    metadata: 1
};
function ListItemText({ children, variant }) {
    const { isTabbable } = useTabbableMode();
    const textContent = (jsx(Text, Object.assign({ variant: textVariants[variant], size: textSizes[variant], lineClamp: textLineClamps[variant] }, { children: children })));
    return isTabbable ? jsx("div", Object.assign({ tabIndex: 0 }, { children: textContent })) : textContent;
}

const flexColumnStyles = "_1fi0ss7";
function ListItemTextLayout({
  primary,
  secondary,
  tertiary,
  overline
}) {
  return jsxs("div", Object.assign({
    className: flexColumnStyles
  }, {
    children: [jsx(ListItemText, Object.assign({
      variant: "overline"
    }, {
      children: overline
    })), jsx(ListItemText, Object.assign({
      variant: "primary"
    }, {
      children: primary
    })), jsx(ListItemText, Object.assign({
      variant: "secondary"
    }, {
      children: secondary
    })), jsx(ListItemText, Object.assign({
      variant: "tertiary"
    }, {
      children: tertiary
    }))]
  }));
}

export { ListItemLayout, ListItemText, ListItemTextLayout };
/*  */
