// @oracle/oraclejet-preact/es/UNSAFE_Link.js
/* @oracle/oraclejet-preact: 14.1.0 */
import { jsx } from 'preact/jsx-runtime';
import "./UNSAFE_Link.css";
import { classNames } from './utils/UNSAFE_classNames.js';
import { usePress } from './hooks/UNSAFE_usePress.js';
import 'preact/hooks';

const styles = {
  default: "_13eej50",
  primary: "_19xkmmq",
  secondary: "m5pa5d",
  standalone: "vmvg4u",
  embedded: "oapnei"
};

const noop = () => {};

function Link({
  href,
  variant = 'default',
  placement = 'standalone',
  target,
  ariaDescribedBy,
  accessibleLabel,
  ariaLabelledBy,
  children,
  onClick
}) {
  const linkClass = classNames([styles.default, styles[placement], variant !== 'default' && styles[variant]]);
  const {
    pressProps
  } = usePress(onClick !== null && onClick !== void 0 ? onClick : noop);
  return jsx("a", Object.assign({
    href: href,
    "aria-describedby": ariaDescribedBy,
    "aria-label": accessibleLabel,
    "aria-labelledby": ariaLabelledBy,
    target: target
  }, pressProps, {
    class: linkClass
  }, {
    children: children
  }));
}

export { Link };
/*  */
