// react-router-outlet/lib/RouterOutlet.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.RouterOutlet = RouterOutlet;
exports.default = void 0;

var _objectWithoutPropertiesLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutPropertiesLoose"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _jsxRuntime = require("react/jsx-runtime");

var _react = require("react");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactRouterDom = require("react-router-dom");

var _isGuarded = _interopRequireDefault(require("./isGuarded"));

var _isRedirect = _interopRequireDefault(require("./isRedirect"));

var _excluded = ["routes", "placeholder"];

function resolveFromProps(route, props) {
  var resolveValue = function resolveValue(val) {
    return typeof val === 'function' ? val(props, route) : val;
  };

  if ((0, _isRedirect.default)(route)) {
    return (0, _extends2.default)({}, route, {
      redirectTo: resolveValue(route.redirectTo)
    });
  }

  if ((0, _isGuarded.default)(route)) return (0, _extends2.default)({}, route, {
    fallback: resolveValue(route.fallback)
  });
  return route;
}

function RouterOutlet(props) {
  var routes = props.routes,
      placeholder = props.placeholder,
      outletProps = (0, _objectWithoutPropertiesLoose2.default)(props, _excluded);

  var redirectJsx = function redirectJsx(route, index) {
    var r = route;
    return (0, _jsxRuntime.jsx)(_reactRouterDom.Redirect, {
      from: r.path,
      exact: r.exact,
      strict: r.strict,
      to: r.redirectTo,
      push: r.push
    }, index);
  };

  var fallbackJsx = function fallbackJsx(route, index) {
    var r = route;
    return (0, _jsxRuntime.jsx)(_reactRouterDom.Redirect, {
      from: r.path,
      exact: r.exact,
      strict: r.strict,
      to: r.fallback,
      push: r.push
    }, index);
  };

  var routeJsx = function routeJsx(route, index) {
    var r = route;
    return (0, _jsxRuntime.jsx)(_reactRouterDom.Route, {
      path: r.path,
      exact: r.exact,
      strict: r.strict,
      render: function render() {
        return (0, _jsxRuntime.jsx)(_react.Suspense, {
          fallback: placeholder || (0, _jsxRuntime.jsx)("div", {}),
          children: (0, _jsxRuntime.jsx)(r.component, (0, _extends2.default)({}, outletProps, r.componentProps, {
            routes: r.routes
          }))
        });
      }
    }, index);
  };

  return (0, _jsxRuntime.jsx)(_reactRouterDom.Switch, {
    children: routes.map(function (route, index) {
      route = resolveFromProps(route, props);
      return (0, _isRedirect.default)(route) ? redirectJsx(route, index) : (0, _isGuarded.default)(route) && !route.canEnter(props, route) ? fallbackJsx(route, index) : routeJsx(route, index);
    })
  });
}

var routePropType = _propTypes.default.shape({
  path: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.arrayOf(_propTypes.default.string)]),
  exact: _propTypes.default.bool,
  strict: _propTypes.default.bool,
  component: _propTypes.default.oneOfType([_propTypes.default.func, _propTypes.default.object]),
  data: _propTypes.default.object,
  componentProps: _propTypes.default.object,
  canEnter: _propTypes.default.func,
  fallback: _propTypes.default.oneOfType([_propTypes.default.func, _propTypes.default.string, _propTypes.default.object]),
  redirectTo: _propTypes.default.oneOfType([_propTypes.default.func, _propTypes.default.string, _propTypes.default.object]),
  push: _propTypes.default.bool
});

RouterOutlet.propTypes = {
  routes: _propTypes.default.arrayOf(routePropType).isRequired,
  placeholder: _propTypes.default.oneOfType([_propTypes.default.func, _propTypes.default.object])
};
var _default = RouterOutlet;
exports.default = _default;
