// @oracle/oraclejet-preact/es/resources/nls/it/bundle.js
/* @oracle/oraclejet-preact: 14.1.0 */
const bundle = {
    close: () => 'Chiudi',
    error: () => 'Errore',
    warn: () => 'Avvertenza',
    info: () => 'Informazioni',
    confirmation: () => 'Conferma',
    progressIndeterminate: () => 'In corso',
    indexer_separatorLabel: (p) => 'Tra ' + p['FROM_SECTION'] + ' e ' + p['TO_SECTION'] + '.',
    indexer_disabledLabel: (p) => p['SECTION'] + '. Nessuna sezione corrispondente trovata.',
    indexer_othersLabel: () => '#',
    indexer_keyboardInstructionText: () => 'Premere Invio per selezionare un valore.',
    indexer_touchInstructionText: () => 'Toccare 2 volte e tenere premuto per immettere la modalità movimento,quindi trascinare su o giù per regolare il valore.',
    inputPassword_show: () => 'Mostra password',
    inputPassword_hide: () => 'Nascondi password',
    inputPassword_hidden: () => 'Password nascosta',
    formControl_clear: () => 'Cancella',
    formControl_loading: () => 'Caricamento',
    formControl_maxLengthExceeded: (p) => 'Lunghezza massima ' + p['MAX_LENGTH'] + ' superata.',
    formControl_maxLengthRemaining: (p) => p['CHARACTER_COUNT'] + ' caratteri rimanenti.',
    formControl_requiredMessageDetail: () => 'Immettere un valore.',
    select_requiredMessageDetail: () => 'Selezionare un valore.',
    select_noMatchesFound: () => 'Nessuna corrispondenza trovata.',
    select_oneMatchFound: () => 'Una corrispondenza trovata',
    select_sizeMatchesFound: (p) => 'Trovate ' + p['TOTAL_SIZE'] + ' corrispondenze',
    select_sizeOrMoreMatchesFound: (p) => 'Trovate ' + p['TOTAL_SIZE'] + ' o più corrispondenze',
    selectMultiple_showSelectedValues: () => 'Mostra solo i valori selezionati.',
    selectMultiple_valuesSelected: (p) => p['VALUE_COUNT'] + ' valori selezionati.',
    selectMultiple_countPlus: (p) => p['COUNT'] + '+',
    selector_selected: () => 'Casella di controllo selezionata',
    selector_unselected: () => 'Casella di controllo non selezionata',
    userAssistance_learnMore: () => 'Ulteriori informazioni',
    userAssistance_required: () => 'Obbligatorio',
    filePicker_singleFileUploadError: () => 'Caricare un solo file alla volta',
    filePicker_singleTypeUploadError: (p) => 'Impossibile caricare i file del tipo ' + p['fileType'],
    filePicker_multipleFileTypeUploadError: (p) => 'Impossibile caricare file di questo tipo: ' + p['fileTypes'],
    filePicker_unknownFileTypeUploadError: () => 'sconosciuto',
    filePicker_dropzonePrimaryText: () => 'Trascina selezione',
    filePicker_dropzoneSecondaryText: () => 'Selezionare un file o rilasciarne uno qui',
    filePicker_dropzoneSecondaryTextMultiple: () => 'Selezionare o rilasciare i file qui',
    filePicker_addFiles: () => 'Aggiungi file',
    plural_separator: () => ', ',
    inputNumber_increase: () => 'Aumenta',
    inputNumber_decrease: () => 'Riduci',
    inputNumber_converterParseError: () => 'Immettere un numero.',
    noData_message: () => 'Nessun elemento da visualizzare',
    message_navigationFromMessagesRegion: () => "Accesso all'area messaggi. Premere F6 per ritornare all'elemento evidenziato in precedenza.",
    message_navigationToMessagesRegion: () => "L'area messaggi contiene nuovi messaggi. Premere F6 per andare all'area messaggi più recenti.",
    messageToast_allMessagesClosed: () => 'Tutti i messaggi popup sono chiusi.'
};

export { bundle as default };
/*  */
