// @oracle/oraclejet-preact/cjs/resources/nls/en-XC/bundle.js
/* @oracle/oraclejet-preact: 14.1.0 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const bundle = {
    close: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::close:::Close]]',
    error: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::error:::Error]]',
    warn: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::warn:::Warning]]',
    info: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::info:::Info]]',
    confirmation: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::confirmation:::Confirmation]]',
    progressIndeterminate: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::progressIndeterminate:::In Progress]]',
    indexer_separatorLabel: (p) => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::indexer_separatorLabel:::Between ' +
        p['FROM_SECTION'] +
        ' and ' +
        p['TO_SECTION'] +
        '.]]',
    indexer_disabledLabel: (p) => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::indexer_disabledLabel:::' +
        p['SECTION'] +
        '. No matching section found.]]',
    indexer_othersLabel: () => '#',
    indexer_keyboardInstructionText: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::indexer_keyboardInstructionText:::Press enter to select value.]]',
    indexer_touchInstructionText: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::indexer_touchInstructionText:::Double tap and hold to enter gesture mode, then drag up or down to adjust value.]]',
    inputPassword_show: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::inputPassword_show:::Show Password]]',
    inputPassword_hide: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::inputPassword_hide:::Hide Password]]',
    inputPassword_hidden: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::inputPassword_hidden:::Password Hidden]]',
    formControl_clear: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::formControl_clear:::Clear]]',
    formControl_loading: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::formControl_loading:::Loading]]',
    formControl_maxLengthExceeded: (p) => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::formControl_maxLengthExceeded:::Maximum length ' +
        p['MAX_LENGTH'] +
        ' exceeded.]]',
    formControl_maxLengthRemaining: (p) => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::formControl_maxLengthRemaining:::' +
        p['CHARACTER_COUNT'] +
        ' characters left.]]',
    formControl_requiredMessageDetail: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::formControl_requiredMessageDetail:::Enter a value.]]',
    select_requiredMessageDetail: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::select_requiredMessageDetail:::Select a value.]]',
    select_noMatchesFound: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::select_noMatchesFound:::No matches found.]]',
    select_oneMatchFound: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::select_oneMatchFound:::One match found]]',
    select_sizeMatchesFound: (p) => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::select_sizeMatchesFound:::' +
        p['TOTAL_SIZE'] +
        ' matches found]]',
    select_sizeOrMoreMatchesFound: (p) => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::select_sizeOrMoreMatchesFound:::' +
        p['TOTAL_SIZE'] +
        ' or more matches found]]',
    selectMultiple_showSelectedValues: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::selectMultiple_showSelectedValues:::Show selected values only.]]',
    selectMultiple_valuesSelected: (p) => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::selectMultiple_valuesSelected:::' +
        p['VALUE_COUNT'] +
        ' values selected.]]',
    selectMultiple_countPlus: (p) => p['COUNT'] + '+',
    selector_selected: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::selector_selected:::Checkbox checked]]',
    selector_unselected: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::selector_unselected:::Checkbox not checked]]',
    userAssistance_learnMore: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::userAssistance_learnMore:::Learn more]]',
    userAssistance_required: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::userAssistance_required:::Required]]',
    filePicker_singleFileUploadError: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::filePicker_singleFileUploadError:::Upload one file at a time]]',
    filePicker_singleTypeUploadError: (p) => "[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::filePicker_singleTypeUploadError:::You can't upload files of type " +
        p['fileType'] +
        ']]',
    filePicker_multipleFileTypeUploadError: (p) => "[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::filePicker_multipleFileTypeUploadError:::You can't upload files of type: " +
        p['fileTypes'] +
        ']]',
    filePicker_unknownFileTypeUploadError: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::filePicker_unknownFileTypeUploadError:::unknown]]',
    filePicker_dropzonePrimaryText: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::filePicker_dropzonePrimaryText:::Drag and Drop]]',
    filePicker_dropzoneSecondaryText: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::filePicker_dropzoneSecondaryText:::Select a file or drop one here]]',
    filePicker_dropzoneSecondaryTextMultiple: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::filePicker_dropzoneSecondaryTextMultiple:::Select or drop files here]]',
    filePicker_addFiles: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::filePicker_addFiles:::Add Files]]',
    plural_separator: () => ', ',
    inputNumber_increase: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::inputNumber_increase:::Increase]]',
    inputNumber_decrease: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::inputNumber_decrease:::Decrease]]',
    inputNumber_converterParseError: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::inputNumber_converterParseError:::Enter a number.]]',
    noData_message: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::noData_message:::No items to display]]',
    message_navigationFromMessagesRegion: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::message_navigationFromMessagesRegion:::Entering messages region. Press F6 to navigate back to prior focused element.]]',
    message_navigationToMessagesRegion: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::message_navigationToMessagesRegion:::Messages region has new messages. Press F6 to navigate to the most recent message region.]]',
    messageToast_allMessagesClosed: () => '[[jet/packages/oraclejet-preact/src/resources/icu/en/bundle.json:::messageToast_allMessagesClosed:::All toast messages are closed.]]'
};

exports["default"] = bundle;
/*  */
