// @oracle/oraclejet-preact/cjs/Badge-c5723bb5.js
/* @oracle/oraclejet-preact: 14.1.0 */
'use strict';

var jsxRuntime = require('preact/jsx-runtime');
require('./utils/UNSAFE_classNames.js');

var classNames = require('./classNames-55bceee6.js');

const styles = {
  base: "_1qv0tr9",
  //right-aligned with the right edge of another component for ltr direction
  //left-aligned for rtl direction pending by
  //https://jira.oraclecorp.com/jira/browse/JET-47572: need an api for rtl
  end: "_10ixplu",
  neutral: "qogxsr",
  danger: "_1raowx1",
  warning: "c3l4at",
  success: "_1gvo3fl",
  info: "_1gbia2f",
  neutralSubtle: "qr381a",
  dangerSubtle: "_61v98d",
  warningSubtle: "_1211o8q",
  successSubtle: "dnak5a",
  infoSubtle: "agx7gh",
  childrenMargins: "_19r40w0"
};
function Badge({
  variant = 'neutral',
  size = 'md',
  edge = 'none',
  children
}) {
  const textClasses = classNames.classNames([styles.base, size === 'sm' && 'oj-badge-sm', edge === 'end' && styles.end, styles[variant], styles.childrenMargins]);
  return jsxRuntime.jsx("span", Object.assign({
    class: textClasses
  }, {
    children: children
  }));
}

exports.Badge = Badge;
/*  */
