// @oracle/oraclejet-preact/cjs/Chip-bc07411b.js
/* @oracle/oraclejet-preact: 14.1.0 */
'use strict';

var jsxRuntime = require('preact/jsx-runtime');
var hooks_UNSAFE_useActionable = require('./hooks/UNSAFE_useActionable.js');

var hooks = require('preact/hooks');
require('./utils/UNSAFE_classNames.js');
var hooks_UNSAFE_useTabbableMode = require('./hooks/UNSAFE_useTabbableMode.js');
var classNames = require('./classNames-55bceee6.js');

const cursorandFocusStyle = "_1u52deh";
const styles = {
  base: "_1tc7kf",
  disabled: "_6h4psl",
  info: "_19774uz",
  infoSubtle: "ks7pum",
  infoSubtleLighter: "soghm",
  childrenMargins: "_7rnbll"
};
function Chip({
  isSelected,
  isDisabled,
  accessibleLabel,
  children,
  onToggle
}) {
  const handleAction = hooks.useCallback(() => {
    onToggle === null || onToggle === void 0 ? void 0 : onToggle({
      previousValue: isSelected,
      value: !isSelected
    });
  }, [isSelected, onToggle]);
  const {
    actionableProps,
    isHover,
    isActive
  } = hooks_UNSAFE_useActionable.useActionable(handleAction);
  const variant = isDisabled ? 'disabled' : isSelected || isActive ? 'info' : isHover ? 'infoSubtle' : 'infoSubtleLighter';
  const classes = classNames.classNames([styles.base, isDisabled ? styles.disabled : `${styles.base} ${styles[variant]} ${cursorandFocusStyle}`, styles.childrenMargins]);
  const {
    isTabbable
  } = hooks_UNSAFE_useTabbableMode.useTabbableMode();
  return isDisabled ? jsxRuntime.jsx("div", Object.assign({
    class: classes
  }, {
    children: children
  })) : jsxRuntime.jsx("div", Object.assign({}, actionableProps, {
    class: classes,
    tabIndex: isTabbable ? 0 : -1,
    "aria-label": accessibleLabel,
    role: "switch",
    "aria-checked": isSelected ? 'true' : 'false'
  }, {
    children: children
  }));
}

exports.Chip = Chip;
/*  */
