// @oracle/oraclejet-preact/cjs/ComponentMessageContainer-7880a1b9.js
/* @oracle/oraclejet-preact: 14.1.0 */
'use strict';

var jsxRuntime = require('preact/jsx-runtime');

require('./PRIVATE_Message.js');
var ComponentMessage = require('./ComponentMessage-eea92e6b.js');
var MessagesManager = require('./MessagesManager-88e137d7.js');

const rootStyles = "_4wirz6";
/**
 * Converts the messages data into Item
 *
 * @param messages The messages data passed down as prop
 * @returns The messages data as Item
 */

function generateMessagesData(messages) {
  return messages.map((message, index) => {
    return {
      key: index,
      data: {
        closeAffordance: 'off',
        severity: message.severity || 'error',
        detail: message.detail
      }
    };
  });
}

function ComponentMessageContainer({
  fieldLabel,
  messages = []
}) {
  return jsxRuntime.jsx("div", Object.assign({
    class: rootStyles
  }, {
    children: jsxRuntime.jsx(MessagesManager.MessagesManager, Object.assign({
      data: generateMessagesData(messages)
    }, {
      children: ({
        item
      }) => jsxRuntime.jsx(ComponentMessage.ComponentMessage, {
        detail: item.data.detail,
        fieldLabel: fieldLabel,
        severity: item.data.severity
      }, item.key)
    }))
  }));
}

exports.ComponentMessageContainer = ComponentMessageContainer;
/*  */
