// @oracle/oraclejet-preact/cjs/ComponentMessage-eea92e6b.js
/* @oracle/oraclejet-preact: 14.1.0 */
'use strict';

var jsxRuntime = require('preact/jsx-runtime');

var hooks_UNSAFE_useTranslationBundle = require('./hooks/UNSAFE_useTranslationBundle.js');
require('./PRIVATE_Message.js');
require('./UNSAFE_HiddenAccessible.js');
require('./utils/UNSAFE_classNames.js');
var classNames = require('./classNames-55bceee6.js');
var MessageUtils = require('./MessageUtils-c82ad817.js');
var HiddenAccessible = require('./HiddenAccessible-42dab376.js');
var MessageStartIcon = require('./MessageStartIcon-5cdb8fbe.js');
var MessageSummary = require('./MessageSummary-ea4ba9b8.js');

const baseStyles = "_11n5zgz";
/**
 * The component that renders an individual message for inline component messaging.
 */

function ComponentMessage({
  detail,
  fieldLabel,
  severity = 'error'
}) {
  const classes = classNames.classNames([baseStyles, MessageUtils.severityBasedStyleClass(severity, 'inline')]);
  const translations = hooks_UNSAFE_useTranslationBundle.useTranslationBundle('@oracle/oraclejet-preact');
  const severityTranslations = {
    confirmation: translations.confirmation(),
    error: translations.error(),
    info: translations.info(),
    warning: translations.warn()
  };
  return jsxRuntime.jsxs("div", Object.assign({
    class: classes,
    role: "alert",
    "aria-atomic": "true"
  }, {
    children: [fieldLabel && jsxRuntime.jsx(HiddenAccessible.HiddenAccessible, {
      children: fieldLabel
    }), MessageUtils.isSeverityIconNeeded(severity) && jsxRuntime.jsx(MessageStartIcon.MessageStartIcon, {
      variant: "inline",
      severity: severity,
      translations: severityTranslations
    }), jsxRuntime.jsx(MessageSummary.MessageSummary, {
      variant: "inline",
      text: detail
    })]
  }));
}

exports.ComponentMessage = ComponentMessage;
/*  */
