// @oracle/oraclejet-preact/cjs/Link-2180b804.js
/* @oracle/oraclejet-preact: 14.1.0 */
'use strict';

var jsxRuntime = require('preact/jsx-runtime');

require('./utils/UNSAFE_classNames.js');
var hooks_UNSAFE_usePress = require('./hooks/UNSAFE_usePress.js');
var classNames = require('./classNames-55bceee6.js');

const styles = {
  default: "_13eej50",
  primary: "_19xkmmq",
  secondary: "m5pa5d",
  standalone: "vmvg4u",
  embedded: "oapnei"
};

const noop = () => {};

function Link({
  href,
  variant = 'default',
  placement = 'standalone',
  target,
  ariaDescribedBy,
  accessibleLabel,
  ariaLabelledBy,
  children,
  onClick
}) {
  const linkClass = classNames.classNames([styles.default, styles[placement], variant !== 'default' && styles[variant]]);
  const {
    pressProps
  } = hooks_UNSAFE_usePress.usePress(onClick !== null && onClick !== void 0 ? onClick : noop);
  return jsxRuntime.jsx("a", Object.assign({
    href: href,
    "aria-describedby": ariaDescribedBy,
    "aria-label": accessibleLabel,
    "aria-labelledby": ariaLabelledBy,
    target: target
  }, pressProps, {
    class: linkClass
  }, {
    children: children
  }));
}

exports.Link = Link;
/*  */
