// @oracle/oraclejet-preact/cjs/MessageCloseButton-66757186.js
/* @oracle/oraclejet-preact: 14.1.0 */
'use strict';

var jsxRuntime = require('preact/jsx-runtime');

require('./UNSAFE_Button.js');
require('./index-3651c0e7.js');
require('./utils/UNSAFE_classNames.js');
var Button = require('./Button-07d97191.js');
var UNSAFE_Icons = require('./index-d5067e50.js');
var classNames = require('./classNames-55bceee6.js');

const messageCloseButtonStyles = {
  base: "bg894j",
  banner: "_1xbtxjt",
  inline: undefined,
  toast: "_1c43iuh"
};
/**
 * A Component for rendering the message close button
 */

function MessageCloseButton({
  onAction,
  buttonRenderer,
  title = 'Close',
  variant = 'banner'
}) {
  const renderedButton = buttonRenderer ? buttonRenderer(title, onAction, variant) : jsxRuntime.jsx(Button.Button, {
    startIcon: jsxRuntime.jsx(UNSAFE_Icons.SvgIcoClose, {}),
    size: "sm",
    title: title,
    variant: "borderless",
    onAction: onAction
  }); // TODO: JET-48715 - remove marker classes
  // Right now, oj-message-banner component uses this for unit tests and has no alternate way for
  // testing. This will need to be removed as we work on oj-c-message-banner.

  const markerClassForTests = 'oj-c-messagebanner-close-icon';
  const classes = classNames.classNames([variant === 'banner' && markerClassForTests, messageCloseButtonStyles.base, messageCloseButtonStyles[variant]]); // Otherwise, render the close icon

  return jsxRuntime.jsx("div", Object.assign({
    class: classes
  }, {
    children: renderedButton
  }));
}

exports.MessageCloseButton = MessageCloseButton;
/*  */
