// @oracle/oraclejet-preact/cjs/Modal-40a059dc.js
/* @oracle/oraclejet-preact: 14.1.0 */
'use strict';

var jsxRuntime = require('preact/jsx-runtime');
var useModal = require('./useModal-0b4e4a2d.js');
require('./UNSAFE_FocusTrap.js');
var UNSAFE_Layer = require('./UNSAFE_Layer.js');

var compat = require('preact/compat');
var FocusTrap = require('./FocusTrap-8af50913.js');

const styles$2 = {
  base: "_52tt52"
};

const ModalBackdrop = () => jsxRuntime.jsx("div", {
  class: styles$2.base
});

const styles$1 = {
  base: "_1mf5n7"
};
const ModalContainer = compat.forwardRef(({
  children
}, ref) => jsxRuntime.jsx("div", Object.assign({
  class: styles$1.base,
  ref: ref
}, {
  children: children
})));
ModalContainer.displayName = 'Forwarded<ModalContainer>';

const styles = {
  base: "zb31o6"
};

const ModalWrapper = ({
  children
}) => jsxRuntime.jsx("div", Object.assign({
  class: styles.base
}, {
  children: children
}));

const Modal = ({
  autoFocusRef,
  children,
  isOpen,
  onBackdropClick,
  restoreFocusRef
}) => {
  const {
    modalRef
  } = useModal.useModal({
    isOpen,
    onBackdropClick
  });
  return isOpen ? jsxRuntime.jsx(UNSAFE_Layer.Layer, {
    children: jsxRuntime.jsxs(ModalWrapper, {
      children: [jsxRuntime.jsx(ModalBackdrop, {}), jsxRuntime.jsx(ModalContainer, Object.assign({
        ref: modalRef
      }, {
        children: jsxRuntime.jsx(FocusTrap.FocusTrap, Object.assign({
          autoFocusRef: autoFocusRef,
          restoreFocusRef: restoreFocusRef
        }, {
          children: children
        }))
      }))]
    })
  }) : null;
};

exports.Modal = Modal;
/*  */
