// @oracle/oraclejet-preact/cjs/Text-cdd6f2fd.js
/* @oracle/oraclejet-preact: 14.1.0 */
'use strict';

var tslib_es6 = require('./tslib.es6-6f60dd09.js');
var jsxRuntime = require('preact/jsx-runtime');

var utils_UNSAFE_mergeInterpolations = require('./utils/UNSAFE_mergeInterpolations.js');

const styles = {
  base: "gdcsb5",
  lineClamp: "_1ml467m",
  truncation: {
    none: '',
    clip: "_12rnn31",
    ellipsis: "bdqm0b"
  },
  variant: {
    primary: "khumtt",
    secondary: "_8znkve",
    disabled: "_1vd9n7v",
    danger: "rx85oh",
    warning: "e3hpcg",
    success: "vq8hn",
    inherit: "_73bp01"
  },
  size: {
    '2xs': "_12f4mgt",
    xs: "v3zzw6",
    sm: "_1mjcipk",
    md: "_1ruobuh",
    lg: "lgtqm3",
    xl: "_1tqa4r9",
    inherit: "_1fx5if7"
  },
  weight: {
    normal: "sn3mkp",
    semiBold: "_8rwin0",
    bold: "_8rddsf",
    inherit: "c401aa"
  },
  hyphens: {
    auto: "_154bi7m",
    none: "q2e634"
  }
};
const interpolations = [() => ({
  class: styles.base
}), ({
  size = 'inherit'
}) => ({
  class: styles.size[size]
}), ({
  weight = 'inherit'
}) => ({
  class: styles.weight[weight]
}), ({
  variant = 'inherit'
}) => ({
  class: styles.variant[variant]
}), ({
  hyphens = 'auto'
}) => ({
  class: styles.hyphens[hyphens]
}), ({
  truncation = 'none'
}) => ({
  class: styles.truncation[truncation]
}), ({
  lineClamp
}) => !lineClamp ? {} : {
  class: styles.lineClamp,
  webkitLineClamp: lineClamp
}];
const styleInterpolations = utils_UNSAFE_mergeInterpolations.mergeInterpolations(interpolations);
const childrenStyles = "_1tb8iy";
function Text(_a) {
  var {
    children
  } = _a,
      props = tslib_es6.__rest(_a, ["children"]);

  const _b = styleInterpolations(props),
        {
    class: cls
  } = _b,
        styles = tslib_es6.__rest(_b, ["class"]);

  return jsxRuntime.jsx("span", Object.assign({
    class: cls,
    style: styles,
    id: props.id
  }, {
    children: jsxRuntime.jsx("span", Object.assign({
      className: childrenStyles
    }, {
      children: children
    }))
  }));
}

exports.Text = Text;
/*  */
