// @oracle/oraclejet-preact/cjs/UNSAFE_Environment.js
/* @oracle/oraclejet-preact: 14.1.0 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var preact = require('preact');
var jsxRuntime = require('preact/jsx-runtime');
var hooks = require('preact/hooks');
var UNSAFE_Layer = require('./UNSAFE_Layer.js');
require('preact/compat');

/**
 * @license
 * Copyright (c) %FIRST_YEAR% %CURRENT_YEAR%, Oracle and/or its affiliates.
 * Licensed under The Universal Permissive License (UPL), Version 1.0
 * as shown at https://oss.oracle.com/licenses/upl/
 * @ignore
 */
var _a, _b;
/**
 * Default environment created for the application
 */
const DefaultEnvironment = {
    user: {
        locale: document.documentElement.getAttribute('lang') || 'en',
        direction: document.documentElement.getAttribute('dir') === 'rtl' ? 'rtl' : 'ltr',
        forcedColors: ((_b = (_a = window.matchMedia) === null || _a === void 0 ? void 0 : _a.call(window, '(forced-colors: active)')) === null || _b === void 0 ? void 0 : _b.matches) ? 'active' : 'none'
    },
    theme: {
        name: 'redwood'
    },
    colorScheme: 'light',
    scale: 'lg'
};
const EnvironmentContext = preact.createContext(DefaultEnvironment);

// Custom merge function
function mergeEnvironment(env1, env2) {
    var _a, _b;
    const userValue = Object.assign({}, env1.user, env2 === null || env2 === void 0 ? void 0 : env2.user);
    const themeValue = Object.assign({}, env1.theme, env2 === null || env2 === void 0 ? void 0 : env2.theme);
    // Merge translation values - one level merge
    const targetTranslations = Object.assign({}, env1.translations);
    const sourceTranslations = (env2 === null || env2 === void 0 ? void 0 : env2.translations) || {};
    Object.keys(sourceTranslations).forEach((key) => {
        let newVal = sourceTranslations[key];
        if (targetTranslations[key]) {
            // merge is needed
            newVal = Object.assign({}, targetTranslations[key], newVal);
        }
        targetTranslations[key] = newVal;
    });
    return {
        user: userValue,
        theme: themeValue,
        translations: targetTranslations,
        colorScheme: (_a = env2 === null || env2 === void 0 ? void 0 : env2.colorScheme) !== null && _a !== void 0 ? _a : env1.colorScheme,
        scale: (_b = env2 === null || env2 === void 0 ? void 0 : env2.scale) !== null && _b !== void 0 ? _b : env1.scale
    };
}
/**
 * The RootEnvironmentProvider component bootstraps essential services and
 * must be placed at the root of any application using components from _'@oracle/oraclejet-preact'_.
 * Minimally, the application must load the appropriate translation bundle for the user's locale
 * and set it on the RootEnvironmentProvider.
 */
function RootEnvironmentProvider({ children, environment }) {
    const mergedEnvironment = hooks.useMemo(() => mergeEnvironment(DefaultEnvironment, environment), [environment]);
    return (jsxRuntime.jsx(EnvironmentContext.Provider, Object.assign({ value: mergedEnvironment }, { children: jsxRuntime.jsx(UNSAFE_Layer.LayerManager, { children: children }) })));
}
/**
 * The EnvironmentProvider is a component that should be used by the application when there is a need to overwrite
 * environment values for a subtree.
 * The component receives an Environment object that will be merged into the values provided by the nearest ancestor Provider.
 * The new environment will be passed to the component's children.
 * Note that some environment values cannot be overwritten. See the description of the Environment type for the list of values
 * that can be replaced.
 */
function EnvironmentProvider({ children, environment }) {
    const consumerEnv = hooks.useContext(EnvironmentContext);
    const mergedEnvironment = hooks.useMemo(() => mergeEnvironment(consumerEnv, environment), [consumerEnv, environment]);
    return (jsxRuntime.jsx(EnvironmentContext.Provider, Object.assign({ value: mergedEnvironment }, { children: children })));
}

exports.EnvironmentContext = EnvironmentContext;
exports.EnvironmentProvider = EnvironmentProvider;
exports.RootEnvironmentProvider = RootEnvironmentProvider;
/*  */
