// @oracle/oraclejet-preact/cjs/UNSAFE_Selector.js
/* @oracle/oraclejet-preact: 14.1.0 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('preact/jsx-runtime');
require("identity-obj-proxy");
require('./utils/UNSAFE_classNames.js');
var utils_UNSAFE_keys = require('./utils/UNSAFE_keys.js');
var hooks_PRIVATE_useSelection = require('./hooks/PRIVATE_useSelection.js');
var hooks_UNSAFE_useTabbableMode = require('./hooks/UNSAFE_useTabbableMode.js');
require('./index-3651c0e7.js');
require('./UNSAFE_Flex.js');
var hooks_UNSAFE_useTranslationBundle = require('./hooks/UNSAFE_useTranslationBundle.js');
var classNames = require('./classNames-55bceee6.js');
var UNSAFE_Icons = require('./index-d5067e50.js');
var Flex = require('./Flex-b113edd5.js');
require('preact/hooks');
require('./utils/UNSAFE_arrayUtils.js');
require('preact');
require('./UNSAFE_Icon.js');
require('./Icon-c44e10c9.js');
require('./tslib.es6-6f60dd09.js');
require('./utils/UNSAFE_size.js');
require('./hooks/UNSAFE_useTooltip.js');
require('./hooks/UNSAFE_useId.js');
require('./UNSAFE_Floating.js');
require('./Floating-48167355.js');
require('preact/compat');
require('./index-55dfe40c.js');
require('./hooks/UNSAFE_useUser.js');
require('./UNSAFE_Environment.js');
require('./UNSAFE_Layer.js');
require('./utils/PRIVATE_floatingUtils.js');
require('./utils/PRIVATE_refUtils.js');
require('./hooks/UNSAFE_useOutsideClick.js');
require('./hooks/UNSAFE_useHover.js');
require('./hooks/UNSAFE_useToggle.js');
require('./hooks/UNSAFE_useFocus.js');
require('./hooks/UNSAFE_useTouch.js');
require('./utils/UNSAFE_mergeProps.js');
require('./hooks/UNSAFE_useAnimation.js');
require('./useAnimation-576ef8bc.js');
require('./hooks/UNSAFE_useTheme.js');
require('./utils/UNSAFE_interpolations/dimensions.js');
require('./_curry1-f1ebd742.js');
require('./utils/UNSAFE_mergeInterpolations.js');
require('./_curry3-bc6e6a86.js');
require('./_curry2-588f7ddc.js');
require('./_has-2eedb7e0.js');
require('./utils/UNSAFE_interpolations/boxalignment.js');
require('./keys-326d80d5.js');
require('./utils/UNSAFE_interpolations/flexbox.js');
require('./flexbox-6deddb70.js');
require('./utils/UNSAFE_interpolations/flexitem.js');
require('./flexitem-0c1e9a72.js');

const styles = {
  base: "_1edubll",
  container: "kx2uew"
};
const checkboxStyles = {
  base: "_1kiq3vp",
  unselected: "_1k5xsej",
  selected: "qtmwhx"
}; // internal component to use in place of the real Checkbox

function Checkbox(props) {
  const {
    isTabbable
  } = hooks_UNSAFE_useTabbableMode.useTabbableMode();
  const classes = classNames.classNames([checkboxStyles.base, props.checked ? checkboxStyles.selected : checkboxStyles.unselected]);
  const translations = hooks_UNSAFE_useTranslationBundle.useTranslationBundle('@oracle/oraclejet-preact');
  const ariaLabel = props.checked ? translations.selector_selected() : translations.selector_unselected();
  return jsxRuntime.jsx("div", Object.assign({
    class: classes,
    tabIndex: isTabbable ? 0 : -1,
    type: "checkbox",
    checked: props.checked,
    "aria-label": props.accessibleLabel || ariaLabel,
    onClick: props.onClick
  }, {
    children: props.checked ? jsxRuntime.jsx(UNSAFE_Icons.SvgCheckboxOn, {}) : jsxRuntime.jsx(UNSAFE_Icons.SvgCheckboxOff, {})
  }));
} // click event handler used by Selector


const handleClick = event => {
  // we don't want container component like ListView to process it
  event.stopPropagation();
}; // update selectionProps onClick event to call event.stopPropagation()


const updateSelectionProps = selectionProps => {
  // click event handler return from useSelection
  const onClick = selectionProps.onClick;

  if (onClick) {
    // click event handler used by Selector
    selectionProps.onClick = event => {
      onClick(event);
      handleClick(event);
    };

    return selectionProps;
  }

  return {
    onClick: handleClick
  };
};
/**
 * The Selector component is used to enable multi-selection in Collection components.
 */


function Selector({
  accessibleLabel,
  rowKey,
  selectedKeys,
  onChange
}) {
  const {
    selectionProps
  } = hooks_PRIVATE_useSelection.useSelection(() => rowKey, selectedKeys, 'multiple', false, 'toggle', onChange); // TODO: replace internal with Preact checkbox component once it is available

  return jsxRuntime.jsx("div", Object.assign({
    class: styles.container
  }, updateSelectionProps(selectionProps), {
    children: jsxRuntime.jsx("div", Object.assign({
      class: styles.base
    }, {
      children: jsxRuntime.jsx(Flex.Flex, Object.assign({
        align: "center",
        justify: "center",
        width: "11x",
        height: "11x"
      }, {
        children: jsxRuntime.jsx(Checkbox, {
          checked: utils_UNSAFE_keys.containsKey(selectedKeys, rowKey),
          accessibleLabel: accessibleLabel
        })
      }))
    }))
  }));
}

/**
 * @license
 * Copyright (c) %FIRST_YEAR% %CURRENT_YEAR%, Oracle and/or its affiliates.
 * The Universal Permissive License (UPL), Version 1.0
 * as shown at https://oss.oracle.com/licenses/upl/
 * @ignore
 */

exports.Selector = Selector;
/*  */
