// @oracle/oraclejet-preact/cjs/UNSAFE_LabelValueLayout.js
/* @oracle/oraclejet-preact: 14.1.0 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('preact/jsx-runtime');
require('./utils/UNSAFE_classNames.js');
require("identity-obj-proxy");
require('./UNSAFE_Flex.js');
var hooks_UNSAFE_useFormContext = require('./hooks/UNSAFE_useFormContext.js');
var hooks_UNSAFE_useFormFieldContext = require('./hooks/UNSAFE_useFormFieldContext.js');
var utils_UNSAFE_size = require('./utils/UNSAFE_size.js');
var classNames = require('./classNames-55bceee6.js');
var Flex = require('./Flex-b113edd5.js');
require('./tslib.es6-6f60dd09.js');
require('./utils/UNSAFE_interpolations/dimensions.js');
require('./utils/UNSAFE_arrayUtils.js');
require('./_curry1-f1ebd742.js');
require('./utils/UNSAFE_mergeInterpolations.js');
require('./_curry3-bc6e6a86.js');
require('./_curry2-588f7ddc.js');
require('./_has-2eedb7e0.js');
require('./utils/UNSAFE_interpolations/boxalignment.js');
require('./keys-326d80d5.js');
require('./utils/UNSAFE_interpolations/flexbox.js');
require('./flexbox-6deddb70.js');
require('./utils/UNSAFE_interpolations/flexitem.js');
require('./flexitem-0c1e9a72.js');
require('preact');
require('preact/hooks');

const labelSlotStyles = {
  base: "cz8tfl",
  start: "a7glsv",
  top: "xuvhwm",
  topPureReadonly: "_19inqct"
};
const labelInnerStyles = {
  // base is labelEdge 'top'
  base: "l53udn",
  start: "cw1gcs",
  startNotPureReadonly: "_9cfba8",
  // TODO: Use Text Component instead of CSS
  noWrap: "_1l3zay2"
};
const valueSlotStyles = "axqmzs";
const LabelValueLayout = ({
  label,
  labelEdge,
  children,
  labelStartWidth = '33%'
}) => {
  const width = utils_UNSAFE_size.sizeToCSS(labelStartWidth);
  const valueWidth = `calc(100% - ${width})`;
  const labelStyles = labelEdge === 'start' ? {
    flexBasis: width,
    width: width,
    maxWidth: width
  } : {};
  const valueStyles = labelEdge === 'start' ? {
    flexBasis: valueWidth,
    width: valueWidth,
    maxWidth: valueWidth
  } : {};
  const {
    isFormLayout,
    isReadonly: isFormReadonly,
    labelWrapping
  } = hooks_UNSAFE_useFormContext.useFormContext();
  const {
    isReadonly
  } = hooks_UNSAFE_useFormFieldContext.useFormFieldContext();
  const isPureReadonly = isFormLayout && isFormReadonly || !isFormLayout && isReadonly;
  const labelStyleClasses = classNames.classNames([labelSlotStyles.base, labelEdge === 'start' && labelSlotStyles.start, // labelEdge === 'start' && isPureReadonly && labelSlotStyles.startReadonly,
  // labelEdge === 'start' && !isPureReadonly && labelSlotStyles.startNotPureReadonly,
  labelEdge === 'top' && labelSlotStyles.top, labelEdge === 'top' && isPureReadonly && labelSlotStyles.topPureReadonly]);
  const labelInnerClasses = classNames.classNames([labelInnerStyles.base, labelEdge === 'start' && labelInnerStyles.start, labelEdge === 'start' && !isPureReadonly && labelInnerStyles.startNotPureReadonly, labelWrapping === 'truncate' && labelInnerStyles.noWrap]);
  return jsxRuntime.jsxs(Flex.Flex, Object.assign({
    wrap: "wrap",
    align: "start"
  }, {
    children: [jsxRuntime.jsx("div", Object.assign({
      class: labelStyleClasses,
      style: labelStyles
    }, {
      children: jsxRuntime.jsx("div", Object.assign({
        class: labelInnerClasses
      }, {
        children: label
      }))
    })), jsxRuntime.jsx("div", Object.assign({
      class: valueSlotStyles,
      style: valueStyles
    }, {
      children: children
    }))]
  }));
};

/**
 * @license
 * Copyright (c) %FIRST_YEAR% %CURRENT_YEAR%, Oracle and/or its affiliates.
 * The Universal Permissive License (UPL), Version 1.0
 * as shown at https://oss.oracle.com/licenses/upl/
 * @ignore
 */

exports.LabelValueLayout = LabelValueLayout;
/*  */
