// @oracle/oraclejet-preact/cjs/UNSAFE_PrefixSuffix.js
/* @oracle/oraclejet-preact: 14.1.0 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('preact/jsx-runtime');
require("identity-obj-proxy");
require('./utils/UNSAFE_classNames.js');
var classNames = require('./classNames-55bceee6.js');

function PrefixSuffix({
  hasEndContent,
  hasInsideLabel,
  hasStartContent,
  id,
  isDisabled,
  isFocused,
  text,
  variant
}) {
  const prefixSuffixStyles = {
    base: "g7fxn4",
    isFocused: "_3r1atv",
    isDisabled: "_13b8ykr",
    insideLabel: "_13lngnj",
    prefixHasStartContent: "k5thct",
    suffixHasEndContent: "_154q0y",
    prefixInlineStart: "_16l446m",
    suffixInlineStart: "_18d8udw"
  }; // We want the input to have padding on itself. This gives the user more space to click on to set focus to the input.
  // The prefix/suffix need padding between itself and either the 'edge' or the 'startContent'/'endContent'.
  // So the logic is:
  // if no startContent and we have a prefix, add --text-to-edge-padding to the prefix
  // if no endContent and we have a suffix, add --text-to-edge-padding to the suffix
  // if startContent and we have a prefix, add text-field-icon-to-text-padding
  // if endContent and we have a suffix, add text-field-icon-to-text-padding
  // **-- text-to-edge-padding**prefix (input has padding for prefix/input)input(input has padding for suffix/input)suffix**--text-to-edge-padding**
  // -- (this should be in startContent)startContent **icon-to-text-padding**prefix (input has padding for prefix/input)input(input has padding for suffix/input) suffix **--text-to-edge-padding**
  // -- **icon-to-text-padding**prefix (input has padding for prefix/input)input(input has padding for suffix/input) suffix**--icon-to-text-padding**(this should be in endContent)endContent

  const prefixSuffixClasses = classNames.classNames([prefixSuffixStyles.base, variant === 'suffix' && hasEndContent && prefixSuffixStyles.suffixHasEndContent, hasInsideLabel && prefixSuffixStyles.insideLabel, variant === 'prefix' && hasStartContent && prefixSuffixStyles.prefixHasStartContent, isFocused && prefixSuffixStyles.isFocused, isDisabled && prefixSuffixStyles.isDisabled, variant === 'prefix' && !hasStartContent && prefixSuffixStyles.prefixInlineStart, variant === 'suffix' && !hasEndContent && prefixSuffixStyles.suffixInlineStart]);
  return jsxRuntime.jsx("div", Object.assign({
    class: prefixSuffixClasses,
    id: id
  }, {
    children: jsxRuntime.jsx("span", {
      children: text
    })
  }));
}

exports.PrefixSuffix = PrefixSuffix;
/*  */
