// @oracle/oraclejet-preact/cjs/hooks/UNSAFE_useTextField.js
/* @oracle/oraclejet-preact: 14.1.0 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var hooks = require('preact/hooks');
var hooks_UNSAFE_useId = require('./UNSAFE_useId.js');

const rankedSeverities = ['error', 'warning', 'confirmation', 'info', 'none'];
/**
 * Merge multiple ids into a space-separated list
 * @param ids The ids to be merged
 * @returns The space separated string of ids
 */
function mergeIDs(...ids) {
    return ids.filter(Boolean).join(' ') || undefined;
}
/**
 * A custom hook to determine the props for a component that renders
 * a text field
 */
function useTextField({ ariaDescribedBy, contentVariant, isDisabled, isFocused, isLoading, isReadonly, labelEdge, messages, styleVariant, value }) {
    const baseId = hooks_UNSAFE_useId.useId();
    const labelId = labelEdge !== 'none' ? `${baseId}-label` : undefined;
    const inputId = !isReadonly ? `${baseId}-input` : undefined;
    const uaId = !(isDisabled || isReadonly) ? `${baseId}-ua` : undefined;
    // Create the form field context
    const formFieldContext = hooks.useMemo(() => {
        const hasValue = value !== undefined && (typeof value !== 'string' || value !== '');
        return isReadonly
            ? {
                hasValue: hasValue,
                isFocused: isFocused,
                isLoading,
                isReadonly: true
            }
            : isDisabled
                ? {
                    hasValue: hasValue,
                    isDisabled: true,
                    isLoading
                }
                : {
                    hasValue: hasValue,
                    isFocused: isFocused,
                    isLoading
                };
    }, [isDisabled, isFocused, isLoading, isReadonly, value]);
    // Determine the highest severity from the messages
    const messageSeverity = hooks.useMemo(() => {
        return messages === undefined
            ? undefined
            : messages.reduce((accSeverity, currMessage) => {
                const currSeverity = currMessage.severity || 'error';
                return rankedSeverities.indexOf(accSeverity) < rankedSeverities.indexOf(currSeverity)
                    ? accSeverity
                    : currSeverity;
            }, 'none');
    }, [messages]);
    // Determine the label variant
    const labelVariant = labelEdge === 'inside'
        ? messageSeverity === 'error'
            ? 'insideError'
            : messageSeverity === 'warning'
                ? 'insideWarning'
                : labelEdge
        : labelEdge !== 'none'
            ? labelEdge
            : undefined;
    return {
        baseId,
        formFieldContext,
        inputProps: {
            id: inputId,
            ariaDescribedBy: mergeIDs(uaId, ariaDescribedBy),
            ariaInvalid: messageSeverity === 'error' ? 'true' : undefined,
            variant: styleVariant
        },
        labelProps: {
            forId: inputId,
            id: labelId,
            variant: labelVariant
        },
        textFieldProps: {
            contentVariant,
            statusVariant: /error|warning/.test(messageSeverity !== null && messageSeverity !== void 0 ? messageSeverity : '')
                ? messageSeverity
                : undefined,
            styleVariant
        },
        userAssistanceProps: {
            id: uaId
        }
    };
}

exports.useTextField = useTextField;
/*  */
