// @oracle/oraclejet-preact/cjs/resources/nls/hu/bundle.js
/* @oracle/oraclejet-preact: 14.1.0 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const bundle = {
    close: () => 'Bezárás',
    error: () => 'Hiba',
    warn: () => 'Figyelmeztetés',
    info: () => 'Információ',
    confirmation: () => 'Megerősítés',
    progressIndeterminate: () => 'Folyamatban',
    indexer_separatorLabel: (p) => p['FROM_SECTION'] + ' és ' + p['TO_SECTION'] + ' között.',
    indexer_disabledLabel: (p) => p['SECTION'] + '. Nem található egyező szakasz.',
    indexer_othersLabel: () => '#',
    indexer_keyboardInstructionText: () => 'Az érték kiválasztásához nyomja meg az Enter billentyűt.',
    indexer_touchInstructionText: () => 'A gesztus módba belépéshez dupla koppintás és megtartás, majd lefelé vagy felfelé húzás az érték beállításához.',
    inputPassword_show: () => 'Jelszó megjelenítése',
    inputPassword_hide: () => 'Jelszó elrejtése',
    inputPassword_hidden: () => 'Jelszó elrejtve',
    formControl_clear: () => 'Törlés',
    formControl_loading: () => 'Betöltés',
    formControl_maxLengthExceeded: (p) => 'Túllépte a maximális hosszt: ' + p['MAX_LENGTH'] + '.',
    formControl_maxLengthRemaining: (p) => p['CHARACTER_COUNT'] + ' karakter maradt.',
    formControl_requiredMessageDetail: () => 'Adjon meg egy értéket.',
    select_requiredMessageDetail: () => 'Jelöljön ki egy értéket.',
    select_noMatchesFound: () => 'Nincs találat.',
    select_oneMatchFound: () => 'Egy egyezés található',
    select_sizeMatchesFound: (p) => p['TOTAL_SIZE'] + ' egyezés található',
    select_sizeOrMoreMatchesFound: (p) => p['TOTAL_SIZE'] + ' vagy több egyezés található',
    selectMultiple_showSelectedValues: () => 'Csak a kijelölt értékek megjelenítése.',
    selectMultiple_valuesSelected: (p) => p['VALUE_COUNT'] + ' kiválasztott érték.',
    selectMultiple_countPlus: (p) => p['COUNT'] + '+',
    selector_selected: () => 'Jelölőnégyzet bejelölve',
    selector_unselected: () => 'Jelölőnégyzet nincs bejelölve',
    userAssistance_learnMore: () => 'További tudnivalók',
    userAssistance_required: () => 'Kötelező',
    filePicker_singleFileUploadError: () => 'Egyszerre csak egy fájlt töltsön fel',
    filePicker_singleTypeUploadError: (p) => 'Nem tölthet fel ' + p['fileType'] + ' típusú fájlt',
    filePicker_multipleFileTypeUploadError: (p) => 'Nem tölthet fel ' + p['fileTypes'] + ' típusú fájlt',
    filePicker_unknownFileTypeUploadError: () => 'ismeretlen',
    filePicker_dropzonePrimaryText: () => 'Áthelyezés húzással',
    filePicker_dropzoneSecondaryText: () => 'Válassza ki a fájlt, vagy húzza át ide',
    filePicker_dropzoneSecondaryTextMultiple: () => 'Válassza ki a fájlokat, vagy húzza át ide',
    filePicker_addFiles: () => 'Fájlok hozzáadása',
    plural_separator: () => ', ',
    inputNumber_increase: () => 'Növelés',
    inputNumber_decrease: () => 'Csökkentés',
    inputNumber_converterParseError: () => 'Adjon meg egy számot.',
    noData_message: () => 'Nincs megjeleníthető elem',
    message_navigationFromMessagesRegion: () => 'Belépés az üzenetek régióba. Az F6 billentyűvel léphet a korábban kijelölt elemhez.',
    message_navigationToMessagesRegion: () => 'Az üzenetek régióban új üzenetek találhatók. Az F6 billentyűvel léphet a legutóbbi üzenetrégióhoz.',
    messageToast_allMessagesClosed: () => 'Az összes információs üzenet bezárva.'
};

exports["default"] = bundle;
/*  */
