// @oracle/oraclejet-preact/es/resources/nls/da/bundle.js
/* @oracle/oraclejet-preact: 14.1.0 */
const bundle = {
    close: () => 'Luk',
    error: () => 'Fejl',
    warn: () => 'Advarsel',
    info: () => 'Oplysninger',
    confirmation: () => 'Bekræftelse',
    progressIndeterminate: () => 'I gang',
    indexer_separatorLabel: (p) => 'Mellem ' + p['FROM_SECTION'] + ' og ' + p['TO_SECTION'] + '.',
    indexer_disabledLabel: (p) => p['SECTION'] + '. Intet matchende afsnit fundet.',
    indexer_othersLabel: () => '#',
    indexer_keyboardInstructionText: () => 'Tryk på Enter for at vælge en værdi.',
    indexer_touchInstructionText: () => 'Dobbelttryk og hold for at aktivere bevægelsestilstand. Træk derefter op eller ned for at justere værdien.',
    inputPassword_show: () => 'Vis adgangskode',
    inputPassword_hide: () => 'Skjul adgangskode',
    inputPassword_hidden: () => 'Adgangskode skjult',
    formControl_clear: () => 'Ryd',
    formControl_loading: () => 'Indlæser',
    formControl_maxLengthExceeded: (p) => 'Maksimumlængden ' + p['MAX_LENGTH'] + ' er overskredet.',
    formControl_maxLengthRemaining: (p) => p['CHARACTER_COUNT'] + ' tegn tilbage.',
    formControl_requiredMessageDetail: () => 'Indtast en værdi.',
    select_requiredMessageDetail: () => 'Vælg en værdi.',
    select_noMatchesFound: () => 'Ingen matchninger blev fundet.',
    select_oneMatchFound: () => 'Én matchning blev fundet',
    select_sizeMatchesFound: (p) => p['TOTAL_SIZE'] + ' matches fundet',
    select_sizeOrMoreMatchesFound: (p) => p['TOTAL_SIZE'] + ' eller flere matches fundet',
    selectMultiple_showSelectedValues: () => 'Vis kun valgte værdier.',
    selectMultiple_valuesSelected: (p) => p['VALUE_COUNT'] + ' værdier valgt.',
    selectMultiple_countPlus: (p) => p['COUNT'] + '+',
    selector_selected: () => 'Afkrydsningsfelt markeret',
    selector_unselected: () => 'Afkrydsningsfelt ikke markeret',
    userAssistance_learnMore: () => 'Lær mere',
    userAssistance_required: () => 'Påkrævet',
    filePicker_singleFileUploadError: () => 'Upload en fil ad gangen',
    filePicker_singleTypeUploadError: (p) => 'Du kan ikke uploade filer af typen ' + p['fileType'],
    filePicker_multipleFileTypeUploadError: (p) => 'Du kan ikke uploade filer af typen: ' + p['fileTypes'],
    filePicker_unknownFileTypeUploadError: () => 'ukendt',
    filePicker_dropzonePrimaryText: () => 'Træk og slip',
    filePicker_dropzoneSecondaryText: () => 'Vælg en fil, eller slip en her',
    filePicker_dropzoneSecondaryTextMultiple: () => 'Vælg eller slip filer her',
    filePicker_addFiles: () => 'Tilføj filer',
    plural_separator: () => ', ',
    inputNumber_increase: () => 'Forøg',
    inputNumber_decrease: () => 'Formindsk',
    inputNumber_converterParseError: () => 'Indtast et tal.',
    noData_message: () => 'Ingen elementer at vise',
    message_navigationFromMessagesRegion: () => 'Åbner meddelelsesområde. Tryk på F6 for at navigere tilbage til tidligere fokuseret element.',
    message_navigationToMessagesRegion: () => 'Meddelelsesområde har nye meddelelser. Tryk på F6 for at navigere til det seneste meddelelsesområde.',
    messageToast_allMessagesClosed: () => 'Alle toast-meddelelser er lukket.'
};

export { bundle as default };
/*  */
